server/jwtauth.js

```diff
@@ -56,13 +56,17 @@ function checkUserPermission(url) {
       var payload = jsonwebtoken.verify(token, config.AUTH_SECRET_KEY);
       var roles = payload.roles;
       if(roles) {
-        return roles.includes("moderator");
+          for (var r in roles) {
+              var role = roles[r]
```

Comment:
roles is supposed to be an array, you can iterate with `of` or a simple for with an index